File: Android/ch09/Traxy/app/src/main/java/edu/gvsu/cis/traxy/model/JournalEntry.java
package edu.gvsu.cis.traxy.model;

import org.parceler.Parcel;

/**
 * Created by dulimarh on 8/1/17.
 */

@Parcel
public class JournalEntry {
    String caption, date;
    double lat, lng;
    int type; // 1:text, 2:photo, 3:audio, 4:video
    String url, thumbnailUrl;

    public JournalEntry() {
        /* this default constructor is required (albeit it is empty) */
    }

    public String getCaption() {
        return caption;
    }

    public String getDate() {
        return date;
    }

    public double getLat() {
        return lat;
    }

    public double getLng() {
        return lng;
    }

    public int getType() {
        return type;
    }

    public String getUrl() {
        return url;
    }

    public String getThumbnailUrl() {
        return thumbnailUrl;
    }

    public void setCaption(String caption) {
        this.caption = caption;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public void setLng(double lng) {
        this.lng = lng;
    }

    public void setType(int type) {
        this.type = type;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public void setThumbnailUrl(String thumbnailUrl) {
        this.thumbnailUrl = thumbnailUrl;
    }
}

File: iOS/ch10/TraxyApp/TraxyApp/JournalEntry.swift
//
//  JournalEntry.swift
//  TraxyApp
//
//  Created by Jonathan Engelsma on 4/10/17.
//  Copyright © 2017 Jonathan Engelsma. All rights reserved.
//

import Foundation

enum EntryType : Int {
    case text = 1
    case photo
    case audio
    case video
}

struct JournalEntry {
    var key : String?
    var type: EntryType?
    var caption : String?
    var url : String
    var thumbnailUrl : String
    var date : Date?
    var lat : Double?
    var lng : Double?
    
    init(key: String?, type: EntryType?, caption: String?, url: String,
         thumbnailUrl: String, date: Date?, lat: Double?, lng: Double?)
    {
        self.key = key
        self.type = type
        self.caption = caption
        
        self.date = date
        self.lat = lat
        self.lng = lng
        self.url = url
        self.thumbnailUrl = thumbnailUrl
    }
}
